### PR TITLE
Add Norgo NGE101 support by jamaron

### PR DESCRIPTION
**Pull request by zuckschwerdt**
S.a. #1002

### CODE REVIEW

**Comment by merbanan** [comment]
Can you change the author of the original commit?

File: src/devices/norgo.c [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+        impulse_gap = (reflect4(b[2] & 0xF) << 0) +
+                ((uint32_t)reverse8(b[3]) << 4) +
+                (((uint32_t)reverse8(b[4]) & 0x7F) << 12);
+        /* clang-format off */
+        data = data_make(
+              "model",      "",             DATA_STRING, "Norgo-NGE101",
+              "id",         "Device ID",    DATA_INT,    device_id,
+              "channel",    "Channel",      DATA_INT,    channel,
+              "gap",        "Impulse gap",  DATA_INT,    impulse_gap,
```
Comment (merbanan) [inline src/devices/norgo.c]: MIC is missing.

File: src/devices/norgo.c [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+        /* Pulse count is totally 34 bits but we report only 32 bits (long int not
+         * supported), which should be enough for the duration of battery */
+        /* clang-format off */
+        data = data_make(
+                "model",        "",             DATA_STRING, "Norgo-NGE101",
+                "id",           "Id",           DATA_INT,    device_id,
+                "channel",      "Channel",      DATA_INT,    channel,
+                "impulses",     "Impulses",     DATA_INT,    (uint32_t)impulses,
+                "battery_ok",   "Battery",      DATA_STRING, !low_battery,
```
Comment (merbanan) [inline src/devices/norgo.c]: MIC is missing.

**Comment by zuckschwerdt** [comment]
Oh, i missed the MIC, thanks! I guess I'll clean this up some more. Then squash and merge manually with different author.

**Comment by merbanan** [comment]
Looks good otherwise imo. Ok to merge.

**Comment by zuckschwerdt** [comment]
I juuust need to take a *quick* peak at the checksum and see if it's an Galois (or Fibonacci) LFSR digest ;)

**Comment by merbanan** [comment]
Yeah the crc looked different. Anyway so far I have only seen, checksums, crc and xorsums. I really doubt it is something else (Occam's razor).

**Comment by merbanan** [comment]
Is there anything missing for this PR?

**Comment by zuckschwerdt** [comment]
Still didn't get around to properly inspect the checksum.

**Comment by jamaron** [comment]
Would it be possible to publish NGE101 support? I haven't seen any problems with the crc calculation that Jesper Hansen had discovered, so that implementation could be used until finding the official one.

**Comment by zuckschwerdt** [comment]
Thanks for the ping. Yes.

**Pull request merged**